An error that appears to happen when a new client joins with existing aircraft

```
[WAC Aircraft] lua/entities/wac_hc_base/cl_init.lua:164: Tried to use a NULL entity!
1. GetPos - [C]:-1
2. attachmentThink - lua/entities/wac_hc_base/cl_init.lua:164
3. unknown - lua/entities/wac_hc_base/cl_init.lua:86
```
wac_hc_base/cl_init.lua:164:
```lua
local localAng = self:WorldToLocalAngles((tr.HitPos - t.model:GetPos()):Angle())
```

This is still present in the current version:
```
[ERROR] lua/entities/wac_hc_base/cl_init.lua:165: Tried to use a NULL entity!
  1. GetPos - [C]:-1
   2. attachmentThink - lua/entities/wac_hc_base/cl_init.lua:165
    3. unknown - lua/entities/wac_hc_base/cl_init.lua:85
```

I'll see what I can do about this.

--- a/WAC Aircraft/lua/entities/wac_hc_base/cl_init.lua
+++ b/WAC Aircraft/lua/entities/wac_hc_base/cl_init.lua
@@ -183,7 +183,7 @@ function ENT:OnRemove()
 			t.model:Remove()
 		end
 	end
-	if IsValid(t.model) then
+	if IsValid(self.Emitter) then
 		self.Emitter:Finish()
 	end
 end

--- a/WAC Aircraft/lua/entities/wac_hc_base/cl_init.lua
+++ b/WAC Aircraft/lua/entities/wac_hc_base/cl_init.lua
@@ -162,11 +162,13 @@ function ENT:attachmentThink()
 	if !camAng then return end
 	local tr = util.QuickTrace(self:LocalToWorld(self.Camera.pos)+camAng:Forward()*20, camAng:Forward()*999999999, self)
 	for _, t in pairs(self.weaponAttachments) do
-		local localAng = self:WorldToLocalAngles((tr.HitPos - t.model:GetPos()):Angle())
-		localAng = Angle(t.restrictPitch and 0 or localAng.p, t.restrictYaw and 0 or localAng.y, t.r or 0)
-		t.model:SetAngles(self:LocalToWorldAngles(localAng))
-		if t.offset then
-			t.model:SetPos(self:LocalToWorld(t.pos) + t.model:LocalToWorld(t.offset) - t.model:GetPos())
+		if IsValid(t.model) then
+			local localAng = self:WorldToLocalAngles((tr.HitPos - t.model:GetPos()):Angle())
+			localAng = Angle(t.restrictPitch and 0 or localAng.p, t.restrictYaw and 0 or localAng.y, t.r or 0)
+			t.model:SetAngles(self:LocalToWorldAngles(localAng))
+			if t.offset then
+				t.model:SetPos(self:LocalToWorld(t.pos) + t.model:LocalToWorld(t.offset) - t.model:GetPos())
+			end
 		end
 	end
 end
@@ -177,9 +179,13 @@ function ENT:OnRemove()
 		s:Stop()
 	end
 	for _, t in pairs(self.weaponAttachments) do
-		t.model:Remove()
+		if IsValid(t.model) then
+			t.model:Remove()
+		end
 	end
-	self.Emitter:Finish()
+	if IsValid(t.model) then
+		self.Emitter:Finish()
+	end
 end
 
 

